privacyidea/lib/apps.py

```diff
@@ -126,7 +133,7 @@ def create_google_authenticator_url(key=None, user=None,
 
     key_bin = binascii.unhexlify(key)
     # also strip the padding =, as it will get problems with the google app.
-    otpkey = base64.b32encode(key_bin).strip('=')
+    otpkey = base64.b32encode(key_bin).decode('utf-8').strip('=')
```

Comment:
i would prefer something like `to_unicode(...)` instead of `.decode()`, i find it more 'literal'. What do You think?

Reply:
hm, I think there are sill a lot of occurrences of "decode".
We should then also change this in the rest of the code.

Reply:
Yeah, ``.decode('utf-8')`` isn't that nice. :-/ But as far as I can see, ``to_unicode`` is more of a Holzhammer function, because it does nothing if the input is already a unicode string, and decodes to a unicode string only if the input is a byte string. But in the case above, we *know* that ``b32encode`` returns a byte string, so we know that we always need to decode. But maybe that's nitpicking -- what do you think?

I was also thinking about creating more literal encoding functions that return unicode strings, similarly to the ``hexlify_and_unicode`` we already have:
```python
def b32encode_and_unicode(s):
    return b32encode(s).decode('utf-8')
```

Reply:
You are right, if the output is known to be a byte string, a following `.decode()` is probably more straight forward.
I like the idea of a helper function like `b32encode_and_unicode()`, maybe also for `b64encode`.

Reply:
I've added a ``b32encode_and_unicode()`` function.